src/main/java/net/sllmdilab/dordriver/generator/AbstractWaveFormData.java

```diff
@@ -0,0 +1,144 @@
+package net.sllmdilab.dordriver.generator;
+
+import java.util.List;
+import java.util.Map;
+
+import net.sllmdilab.commons.util.T5FHIRUtils;
+import net.sllmdilab.dordriver.domain.Mapable;
+
+public abstract class AbstractWaveFormData implements Mapable {
+	public static final String RANGE_HIGH_KEY = "RANGE_HIGH";
+	public static final String RANGE_HIGH_PREFIX = RANGE_HIGH_KEY + "_";
+	public static final String RANGE_LOW_KEY = "RANGE_LOW";
+	public static final String RANGE_LOW_PREFIX = RANGE_LOW_KEY + "_";
+	public static final String RATE_KEY = "RATE";
+	public static final String RATE_PREFIX = RATE_KEY + "_";
+	public static final String SAMPLES_KEY = "SAMPLES";
+	public static final String SAMPLES_PREFIX = SAMPLES_KEY + "_";
+	public static final String START_TIME_KEY = "START_TIME";
+	public static final String END_TIME_KEY = "END_TIME";
+	public static final String DATA_TYPE_KEY = "DATA_TYPE";
+
+	private List<Double> samples;
+	private double rangeLow;
+	private double rangeHigh;
+	private double rate;
+	private String startTime;
+	private String endTime;
+	private WaveFormType type;
+
+	private String sampleSeparator = "^";
+
+	public AbstractWaveFormData() {
+
+	}
+
+	public AbstractWaveFormData(WaveFormType type) {
+		this.type = type;
+	}
+
+	public AbstractWaveFormData(List<Double> samples, double rangeLow, double rangeHigh, double rate, WaveFormType type) {
+		super();
```

Comment:
Man skulle kunna kalla på den andra konstruktorn med this() här eller från den andra konstruktorn.